### PR TITLE
DSG-2556: split file upload component into single and variable

### PR DESCRIPTION
**Pull request by kevincheng0**
# Basics
Split file upload component into single and variable file upload

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
- Added subfolders to the file-upload folder in `form-nuv/src/lib/components/advanced`
- `single-file-upload` and `variable-file-upload` both have formio and formly subfolders. Each one depends on a shared `templates` folder to reuse styles and html templates.
- Restrict variable file upload component to only one row of document
- Fixed key selector wrapping in document table
- Added validation for keys to exist

## Screenshot of the affected changes
image
image
image
image
image

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/shared/feature/form-nuv/src/lib/components/advanced/file-upload/variable-file-upload/formio/formio-variable-file-upload.component.spec.ts
Comment (thechechogomez) [inline libs/shared/feature/form-nuv/src/lib/components/advanced/file-upload/variable-file-upload/formio/formio-variable-file-upload.component.spec.ts:1]: This file should be ignored by sonarqube. I think we can delete it.

File: libs/shared/feature/form-nuv/src/lib/components/advanced/file-upload/variable-file-upload/formio/formio-variable-file-upload.model.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+};
+
+const COMPONENT_OPTIONS: FormioCustomComponentInfo = {
+  ...DEFAULT_COMPONENT_OPTIONS,
+  editForm, // Optional: define the editForm of the field. Default: the editForm of a textfield
+  group: 'nuverialAdvanced', // Build Group
+  icon: 'file', // Icon
+  schema,
+  selector, // custom selector. Angular Elements will create a custom html tag with this selector
+  title: 'Variable File Upload', // Title of the component
```
Comment (thechechogomez) [inline libs/shared/feature/form-nuv/src/lib/components/advanced/file-upload/variable-file-upload/formio/formio-variable-file-upload.model.ts:40]: Don't know if this is the best name possible. Having variable on the multiple file upload while being able to add multiple documents on a single file upload is kind of confusing. Maybe change this name to Multiple File Upload?

**Review by thechechogomez — COMMENTED** [review]
Looking good. Left a couple of comments. Additionally, I can save a File Upload component on the frontend without a key. The backend does stop me from saving the form without a key but it would be great to have that validation on the frontend as well.

File: libs/shared/feature/form-nuv/src/lib/components/advanced/file-upload/variable-file-upload/formio/formio-variable-file-upload.model.ts (hunk 1 above)
Comment (kevincheng0) [reply libs/shared/feature/form-nuv/src/lib/components/advanced/file-upload/variable-file-upload/formio/formio-variable-file-upload.model.ts:41]: Yup, I'll bring it up during standup tomorrow

**Review by thechechogomez — APPROVED** [review]
Looks great!

**Review by thechechogomez — APPROVED** [review]

**Pull request merged**